noir_stdlib/src/hash/pedersen.nr

```diff
@@ -0,0 +1,24 @@
+use crate::hash::{Hasher, pedersen_hash};
+use crate::default::Default;
+
+struct PedersenHasher{
+    _state: [Field]
+}
+
+impl Hasher for PedersenHasher {
+    fn finish(self) -> Field {
+       pedersen_hash(self._state)
+    }
+
+    fn write(&mut self, input: [Field]){
+        self._state = self._state.append(input);
+    }
+}
+
+impl Default for PedersenHasher{
+    fn default() -> Self{
+        PedersenHasher{
+            _state: []
```

Comment:
Flagging this for @vezenovm to comment on if there are any known existing bugs with slices that'd affect the usecase of implementing a Hasher like this.

Reply:
This should be fine as it accepts a slice of primitive types (Fields in this case). An empty slice will still hash without error.